readthedocs/doc_builder/environments.py

```diff
@@ -301,11 +301,35 @@ def run(self):
 
         self.start_time = datetime.utcnow()
         client = self.build_env.get_client()
+
+        # Create a copy of the environment to update PATH variable
+        environment = self._environment.copy()
+        # Default PATH variable
+        environment[
+            "PATH"
+        ] = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+        # Add asdf extra paths
+        environment["PATH"] += (
+            f":/home/{settings.RTD_DOCKER_USER}/.asdf/shims"
+            f":/home/{settings.RTD_DOCKER_USER}/.asdf/bin"
+        )
+
+        if settings.RTD_DOCKER_COMPOSE:
+            environment["PATH"] += ":/root/.asdf/shims:/root/.asdf/bin"
+
+        # Prepend the BIN_PATH if it's defined
+        if self.bin_path:
+            path = environment.get("PATH")
+            bin_path = self._escape_command(self.bin_path)
```

Comment:
Why are we only escaping this here? If this is user input and security critical, it should be very explicitly mentioned.

Reply:
I'm not sure why that `_escape_command` is there. I copied from our previous code,

https://github.com/readthedocs/readthedocs.org/blob/830b899c91ab9f0ad8f8448947461d1eeafa23a1/readthedocs/doc_builder/environments.py#L369-L371

In theory, it's not input from the user. This should be the venv path, if I understand correctly this code.

Reply:
It may be not required, I think 🤷🏼 

After a quick grep, it's always `self.python_env.venv_bin()` or `self.venv_bin()`. So, we could probably remove it if we want, but I'm not sure.